--- subtitleeditor-0.36.1.orig/plugins/actions/scalesubtitles/dialog-scale-subtitles.ui	2010-01-19 13:04:21.000000000 +0100
+++ subtitleeditor-0.36.1/plugins/actions/scalesubtitles/dialog-scale-subtitles.ui	2010-06-05 20:53:53.318180804 +0200
@@ -342,6 +342,20 @@
                     <property name="position">1</property>
                   </packing>
                 </child>
+
+                        <child>
+                          <object class="GtkCheckButton" id="check-all-subtitles">
+                            <property name="visible">True</property>
+                            <property name="can_focus">True</property>
+                            <property name="tooltip_text">If the option is activated, all subtitles are scaled. Without this option, it scales only subtitles between the two selected points.</property>
+                            <property name="label" translatable="yes">_All subtitles</property>
+                            <property name="use_underline">True</property>
+                            <property name="response_id">0</property>
+                            <property name="draw_indicator">True</property>
+                          </object>
+                        </child>
+
+                
               </object>
               <packing>
                 <property name="position">1</property>
--- subtitleeditor-0.36.1.orig/plugins/actions/scalesubtitles/scalesubtitles.cc	2009-12-21 12:49:24.000000000 +0100
+++ subtitleeditor-0.36.1/plugins/actions/scalesubtitles/scalesubtitles.cc	2010-06-05 20:57:52.044930839 +0200
@@ -50,6 +50,7 @@
 		builder->get_widget_derived("spin-last-start-value", m_spinLastStartValue);
 		builder->get_widget_derived("spin-last-new-start", m_spinLastNewStart);
 		builder->get_widget("label-last-text", m_labelLastText);
+		builder->get_widget("check-all-subtitles", m_allSelectedSubtitles);
 
 		// signaux
 		m_spinFirstNumber->signal_value_changed().connect(
@@ -93,10 +94,12 @@
 				SubtitleTime dest1( (long int)m_spinFirstNewStart->get_value() );
 				SubtitleTime dest2( (long int)m_spinLastNewStart->get_value() );
 
+				bool all_subtitles = m_allSelectedSubtitles->get_active();
+
 				// apply change
 				doc->start_command(_("Scale subtitles"));
 				
-				scale(firstSubtitle, dest1, lastSubtitle, dest2);
+				scale(doc, all_subtitles, firstSubtitle, dest1, lastSubtitle, dest2);
 
 				doc->emit_signal("subtitle-time-changed");
 				doc->finish_command();
@@ -224,6 +227,7 @@
 	 *
 	 */
 	void scale(
+			Document *doc, bool all_subtitles,
 			const Subtitle &sub1, const SubtitleTime &dest1,
 			const Subtitle &sub2, const SubtitleTime &dest2)
 	{
@@ -234,10 +238,14 @@
 				sub1.get_start().totalmsecs, dest1.totalmsecs,
 				sub2.get_start().totalmsecs, dest2.totalmsecs);
 
-		// sub2 + 1
-		Subtitle end = sub2; ++end;
+		Subtitles subtitles = doc->subtitles();
+		// first
+		Subtitle start = (all_subtitles ? subtitles.get_first() : sub1);
+		// last + 1
+		Subtitle end = (all_subtitles ? subtitles.get_last() : sub2);
+		++end;
 
-		for(Subtitle subtitle=sub1; subtitle != end; ++subtitle)
+		for(Subtitle subtitle=start; subtitle != end; ++subtitle)
 		{
 			SubtitleTime start = calcul(subtitle.get_start(), scale, source1, dest1);
 			SubtitleTime end = calcul(subtitle.get_end(), scale, source1, dest1);
@@ -279,6 +287,7 @@
 	Gtk::Label*				m_labelLastStartValue;
 	SpinButtonTime*		m_spinLastNewStart;
 	Gtk::Label*				m_labelLastText;
+	Gtk::CheckButton*	m_allSelectedSubtitles;
 };
 
 /*
