Index: rss_fetch.inc
===================================================================
--- rss_fetch.inc	(revision 556)
+++ rss_fetch.inc	(working copy)
@@ -1,4 +1,4 @@
-<?php
+<?php if (!defined('PmWiki')) exit();
 /*
  * Project:     MagpieRSS: a simple RSS integration tool
  * File:        rss_fetch.inc, a simple functional interface
@@ -460,4 +460,4 @@
     return $sc >= 500 && $sc < 600; 
 }
 
-?>
+
Index: rss_utils.inc
===================================================================
--- rss_utils.inc	(revision 556)
+++ rss_utils.inc	(working copy)
@@ -1,4 +1,4 @@
-<?php
+<?php if (!defined('PmWiki')) exit();
 /*
  * Project:     MagpieRSS: a simple RSS integration tool
  * File:        rss_utils.inc, utility methods for working with RSS
@@ -64,4 +64,4 @@
     }
 }
 
-?>
+
Index: rss_cache.inc
===================================================================
--- rss_cache.inc	(revision 556)
+++ rss_cache.inc	(working copy)
@@ -1,4 +1,4 @@
-<?php
+<?php if (!defined('PmWiki')) exit();
 /*
  * Project:     MagpieRSS: a simple RSS integration tool
  * File:        rss_cache.inc, a simple, rolling(no GC), cache 
@@ -197,4 +197,3 @@
 
 }
 
-?>
Index: rss_parse.inc
===================================================================
--- rss_parse.inc	(revision 556)
+++ rss_parse.inc	(working copy)
@@ -1,4 +1,4 @@
-<?php
+<?php if (!defined('PmWiki')) exit();
 
 /**
 * Project:     MagpieRSS: a simple RSS integration tool
@@ -602,4 +602,3 @@
 
 }
 
-?>
