--- rss_fetch.inc	2005-11-05 22:10:56.000000000 +0100
+++ rss_fetch.inc	2011-11-19 21:57:10.000000000 +0100
@@ -270,6 +270,11 @@
     $client->agent = MAGPIE_USER_AGENT;
     $client->read_timeout = MAGPIE_FETCH_TIME_OUT;
     $client->use_gzip = MAGPIE_USE_GZIP;
+    if(defined('MAGPIE_PROXY')) {
+      list($phost, $pport) = preg_split('/:/', MAGPIE_PROXY);
+      $client->proxy_host = $phost;
+      $client->proxy_port = $pport;
+    } 
     if (is_array($headers) ) {
         $client->rawheaders = $headers;
     }
