--- work/vboxdrv/linux/SUPDrv-linux.c.orig	2009-05-04 13:05:46.000000000 -0700
+++ work/vboxdrv/linux/SUPDrv-linux.c	2009-05-04 13:06:10.000000000 -0700
@@ -769,8 +769,8 @@
     rc = supdrvCreateSession(&g_DevExt, (PSUPDRVSESSION *)&pSession);
     if (!rc)
     {
-        pSession->Uid       = current->euid;
-        pSession->Gid       = current->egid;
+        pSession->Uid       = current->cred->euid;
+        pSession->Gid       = current->cred->egid;
         pSession->Process   = RTProcSelf();
         pSession->R0Process = RTR0ProcHandleSelf();
     }
