--- dbvis	2009-12-13 12:50:32.000000000 +0100
+++ dbvis	2009-12-13 12:51:23.000000000 +0100
@@ -1,5 +1,7 @@
 #!/bin/sh
 
+cd /opt/DbVisualizer/
+
 # Uncomment the following line to override the JVM search sequence
 # INSTALL4J_JAVA_HOME_OVERRIDE=
 # Uncomment the following line to add additional VM parameters
@@ -229,7 +231,7 @@
 progname=`basename "$prg"`
 cd "$prg_dir"
 prg_dir=`pwd`
-app_home=.
+app_home=/opt/DbVisualizer
 cd "$app_home"
 app_home=`pwd`
 bundled_jre_home="$app_home/jre"
