--- inodeproc.cpp	2004-09-14 13:35:54.000000000 +0200
+++ inodeproc.cpp	2008-10-06 17:55:54.000000000 +0200
@@ -10,6 +10,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
+#include <limits.h>
 
 struct proginfo
 {
--- cui.cpp	2004-09-16 19:31:33.000000000 +0200
+++ cui.cpp	2008-10-06 17:59:24.000000000 +0200
@@ -6,6 +6,7 @@
 #include <ncurses.h>
 #include "nethogs.h"
 #include "process.h"
+#include <stdlib.h>
 
 std::string * caption;
 //extern char [] version;
