--- XML-LibXML-1.63/t/03doc.t	2006-09-06 14:11:03.000000000 +0200
+++ XML-LibXML-1.63/t/03doc.t	2007-09-25 10:23:38.000000000 +0200
@@ -206,8 +206,11 @@
         ok($pi);
         ok($pi->nodeType, XML_PI_NODE);
         ok($pi->nodeName, "foo");
-        ok( $pi->textContent, undef);
-        ok( $pi->getData, undef);
+	my $data = $pi->textContent;
+	# undef or "" depending on libxml2 version
+        ok( !defined $data or length($data)==0 );
+	$data = $pi->getData;
+        ok( !defined $data or length($data)==0 );
 	$pi->setData(q(bar&amp;));
 	ok( $pi->getData, q(bar&amp;));
         ok($pi->textContent, q(bar&amp;));
