Fix for gentoo bug #188911

--- a/fdi/fdi.dtd	2007-08-21 18:06:09.000000000 +0200
+++ b/fdi/fdi.dtd	2007-08-21 18:06:24.000000000 +0200
@@ -13,6 +13,7 @@
     key              CDATA #REQUIRED
     string           CDATA #IMPLIED
     int              CDATA #IMPLIED
+    uint64           CDATA #IMPLIED
     bool             (false|true) #IMPLIED
     exists           (false|true) #IMPLIED
     empty            (false|true) #IMPLIED
